components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -230,13 +231,35 @@ export default class ChannelList extends React.PureComponent<ChannelListProps, C
         const {team_ids: teamIds} = filterOptions.teams.values;
         if (publicChannels.value || privateChannels.value || deleted.value || groupConstrained.value || excludeGroupConstrained.value || (teamIds.value as string[]).length) {
             filters.public = publicChannels.value as boolean;
+            if (filters.public) {
+                trackEvent('admin_channels_page', 'public_filter_applied_to_channel_list');
+            }
+
             filters.private = privateChannels.value as boolean;
+            if (filters.private) {
+                trackEvent('admin_channels_page', 'private_filter_applied_to_channel_list');
+            }
+
             filters.deleted = deleted.value as boolean;
+            if (filters.deleted) {
+                trackEvent('admin_channels_page', 'archived_filter_applied_to_channel_list');
+            }
+
             if (!(groupConstrained.value && excludeGroupConstrained.value)) {
                 filters.group_constrained = groupConstrained.value as boolean;
+                if (filters.group_constrained) {
+                    trackEvent('admin_channels_page', 'group_sync_filter_applied_to_channel_list');
+                }
                 filters.exclude_group_constrained = excludeGroupConstrained.value as boolean;
+                if (filters.group_constrained) {
+                    trackEvent('admin_channels_page', 'manual_invites_filter_applied_to_channel_list');
+                }
             }
+
             filters.team_ids = teamIds.value as string[];
+            if (filters.team_ids.length > 0) {
+                trackEvent('admin_channels_page', 'team_id_filter_applied_to_channel_list');
+            }
```

Comment:
Another way to do this would be a filter-condition to tracking-event-name data structure, like:
```js
[
    [(f) => f.public, 'public_filter_applied_to_channel_list'],
    [(f) => f.private, 'private_filter_applied_to_channel_list'],
    [(f) => f.deleted, 'archived_filter_applied_to_channel_list'],
    [(f) => f.group_constrained, 'group_sync_filter_applied_to_channel_list'],
    [(f) => f.exclude_group_constrained, 'manual_invites_filter_applied_to_channel_list'],
    [(f) => f.team_ids.length, 'team_id_filter_applied_to_channel_list'],
].forEach(([condition, eventName]) => {
    if (condition(filters)) {
        trackEvent('admin_channels_page', eventName);
    }
});
```

Reply:
thats nice, but it doesnt play too well with typescript, I could do it with an object probably instead but then i would have to define an inline type that would just end up being more lines of code than the multiple conditions I have right now